cli/src/lib/website/filesInit.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  Args,
+  Mas,
+  MAX_GAS_CALL,
+  minBigInt,
+  Operation,
+  SmartContract,
+  U32,
+} from '@massalabs/massa-web3'
+import { storageCostForEntry } from '../utils/storage'
+import { FileInit } from './models/FileInit'
+import { Metadata } from './models/Metadata'
+import {
+  fileChunkCountKey,
+  fileLocationKey,
+  globalMetadataKey,
+} from './storageKeys'
+import { FileDelete } from './models/FileDelete'
+
+const functionName = 'filesInit'
+const batchSize = 20
+
+/**
+ * Send the filesInits to the smart contract
+ * @param sc - SmartContract instance
+ * @param files - Array of FileInit instances
+ * @param filesToDelete - Array of FileInit instances to delete
+ * @param metadatas - Array of Metadata instances
+ * @returns - Array of Operation instances
+ */
+export async function sendFilesInits(
+  sc: SmartContract,
+  files: FileInit[],
+  filesToDelete: FileDelete[],
+  metadatas: Metadata[],
+  metadatasToDelete: Metadata[]
+): Promise<Operation[]> {
+  const chunkBatches: FileInit[][] = []
```

Comment:
I don't know if we should rename chunkBatches by FilesInit for the comprehension ?